avocado/utils/memory.py

```diff
@@ -320,3 +320,56 @@ def get_thp_value(feature):
         return (re.search(r"\[(\w+)\]", value)).group(1)
     else:
         return value
+
+
+class MemInfo(object):
+    """
+    Representation of /proc/meminfo
+    """
+
+    class MemInfoItem(object):
```

Comment:
Btw I agree the DataSize might come handy, but it's not that complex, nor is it much needed anywhere else, so I'd just keep it. The move to top level is IMO also better, but I considered it your style so did not complain much about it 😉.